net/sched/act_skbedit.c

```diff
@@ -50,6 +53,30 @@ static int tcf_skbedit(struct sk_buff *skb, const struct tc_action *a,
 	}
 	if (d->flags & SKBEDIT_F_PTYPE)
 		skb->pkt_type = d->ptype;
+	if (d->flags & SKBEDIT_F_INHERITDSFIELD) {
+		int wlen = skb_network_offset(skb);
+
+		switch (tc_skb_protocol(skb)) {
+		case htons(ETH_P_IP):
+			wlen += sizeof(struct iphdr);
+			if (!pskb_may_pull(skb, wlen)) {
+				spin_unlock(&d->tcf_lock);
+				return TC_ACT_SHOT;
+			}
+			skb->priority = ipv4_get_dsfield(ip_hdr(skb)) >> 2;
+			break;
+
+		case htons(ETH_P_IPV6):
+			wlen += sizeof(struct ipv6hdr);
+			if (!pskb_may_pull(skb, wlen)) {
+				spin_unlock(&d->tcf_lock);
+				return TC_ACT_SHOT;
+			}
+			skb->priority = ipv6_get_dsfield(ipv6_hdr(skb)) >> 2;
+			break;
+		}
```

Comment:
Add a default clause with the following code:
```
if (d->flags & SKBEDIT_F_PRIORITY)
	skb->priority = d->priority;
break;
```

This will make `d->priority` act as a default priority.